src/java/org/apache/cassandra/cql3/Operator.java

```diff
@@ -248,31 +404,31 @@ public boolean isSatisfiedBy(AbstractType<?> type, ByteBuffer leftOperand, ByteB
     LIKE(14)
```

Comment:
Shouldn't the `LIKE_MATCHES` above be an equality instead?

Reply:
It effectively looks weird to me but I kept the original logic.

Reply:
And the original logic had a bug so let's fix it?

Reply:
I would prefer to do it in another patch. Otherwise, it becomes the responsibility of this patch to test it.
LIKE is a big story in itself and I prefer to avoid being involve in it at this point.

Reply:
Created CASSANDRA-19574